src/components/Modal.js

```diff
@@ -1,14 +1,53 @@
 import React from 'react';
 import '../App.css';
 
-const Modal = props => {
-    return (
-        <div className="Modal">
-            <p> Are you sure you want to delete {props.item.itemName}?</p>
-            <button onClick={() => props.delete(props.item)}>YES</button>
-            <button onClick={() => props.cancel()}>No</button>
-        </div>
-    );
+const Modal = ({ type, item, deleteItem, cancelItem, setDetailModal }) => {
+    let output;
+
+    if (type === 'deleteItem') {
+        output = (
+            <div className="Modal">
+                <p> Are you sure you want to delete {item.itemName}?</p>
+                <button onClick={() => deleteItem(item)}>YES</button>
+                <button onClick={() => cancelItem()}>No</button>
+            </div>
+        );
+    } else if (type === 'detail') {
+        output = (
+            <div className="detailsModal">
+                <h1>Purchase Details</h1>
+                <h2>{item.name}</h2>
```

Comment:
```suggestion
                <h2>{item.itemName}</h2>
```
The name isn't showing up in the modal...I think it needs to be `item.itemName`